v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/transformer/GenericRecordTypeConvertor.java

```diff
@@ -0,0 +1,423 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.transformer;
+
+import com.google.cloud.ByteArray;
+import com.google.cloud.Date;
+import com.google.cloud.spanner.Value;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.ISchemaMapper;
+import com.google.cloud.teleport.v2.spanner.type.Type;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.nio.ByteBuffer;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+import org.apache.avro.Conversions;
+import org.apache.avro.LogicalTypes;
+import org.apache.avro.Schema;
+import org.apache.avro.data.TimeConversions;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Convertor Class containing methods for type conversion of various AvroTypes to Spanner {@link
+ * Value} types.
+ */
+public class GenericRecordTypeConvertor {
+  private static final Logger LOG = LoggerFactory.getLogger(GenericRecordTypeConvertor.class);
+
+  /**
+   * This method takes in a generic record and returns a map between the Spanner column name and the
+   * corresponding Spanner column value. This handles the data conversion logic from a GenericRecord
+   * field to a spanner Value.
+   */
+  public static Map<String, Value> transformChangeEvent(
```

Comment:
We can do either of the approaches you mentioned for namespace, it will come from the source record and we will use schemaMapper to find the spanner namespace (whenever it is released). I lean towards the constructor approach since the method signatures are lighter then.